server/src/main/java/org/opensearch/search/query/QueryCollectorArguments.java

```diff
@@ -0,0 +1,56 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.search.query;
+
+import org.opensearch.common.annotation.ExperimentalApi;
+
+/**
+ * Arguments for {@link QueryCollectorContextSpecRegistry}
+ */
+@ExperimentalApi
+public final class QueryCollectorArguments {
```

Comment:
it's better to add implementations of equals/hashCode for proper behavior in collections